src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -202,8 +200,6 @@ def _make_candidate_from_link(
                     self._build_failures[link] = e
                     return None
                 except InstallationSubprocessError as e:
-                    if not self._suppress_build_failures:
```

Comment:
`_suppress_build_failures` used to default to `False`, so this used to `raise` due to the negation.

As such, it seems the `raise` should be left here, and the other codepath removed instead?

Reply:
Indeed. We should remove this `except` block -- not just the conditional.